src/web/components/SharingPermission/bulkAddPermissionsHelpers.ts

```diff
@@ -40,6 +48,36 @@ export const getRecommendedTypeFromParticipant = (participantTypeNames: string[]
   return formatStringsWithSeparator(result);
 };
 
+export const hasUncheckedDefaultParticipantTypeRecommendation = (
```

Comment:
In my understanding, we only show the warning if user is sharing with the type and now they uncheck any of the value?

Reply:
yes, with the assumption that the ones that are checked are checked because they align with the recommended type given their participant type. So essentially only if they are already sharing with the recommended types and uncheck one of those recommended types.

Reply:
Hmmm could we double check with Kimberly about this? So if we are having a participant currently sharing with a type which is not a recommanded type. and later, they deceided to uncheck the checkbox, are we showing this or not?

Reply:
yep I will double check!

Reply:
The more I think about it I think I have probably misinterpreted the design - I think we probably just want to show the warning whenever we remove any type, but I have asked on Figma anyway.

Reply:
This brings up another question which I have asked in Figma:
let's say this participant is a publisher and has shared with Advertisers and DSPs (despite DSPs being the only recommended type). When the user sees this screen, will DSPs AND Advertisers be checked here? Or just DSPs since that is the recommended type?

Reply:
In my understanding, the recommended component only shows when this participant hasn't shared with any specific participant or type. 
When a participant shares with a specific participant or type (which means now we have that sharing permission table and the top 2 components get collapsed), this bulk add component should only show the current type they are sharing with, in your case both DSPs and Advertisers should be checked

Reply:
yes you are right. I will address this in a future MR.